src/core/vdom/vnode.js

```diff
@@ -19,6 +19,7 @@ export default class VNode {
   isComment: boolean; // empty comment placeholder?
   isCloned: boolean; // is a cloned node?
   isOnce: boolean; // is a v-once node?
+  isAsyncPlaceholder: boolean; // placeholder node for async component
```

Comment:
We have a lot of boolean flags here. I think they can be packed into an integer and be retrieved by bitwise operation. Fewer properties should reduces memory consumption since typically we have a lot of VNode instances and probably can prevent multiple hidden class creation.

Maybe a good refactor for another pull request.